lib/truffle/java/java_proxy.rb

```diff
@@ -25,5 +28,16 @@ def to_java_object
 
   def eql?(another)
   end
-  
+
+  def self.[](*dims)
+    ArrayJavaProxyCreator.new(java_class, *dims)
+  end
+
+  def self.inherited(thing)
+    thing.__send__(:define_singleton_method, :inherited, method(:inherited))
+    unless Thread.current[:MAKE_PROXY]
+      p "Creating #{thing} inherited from #{self} outside the normal make_proxy mechanism."
```

Comment:
Yes. For the moment I've found it extremely helpful to see when gems are attempting to subclass Java classes, but don't want to raise an exception that would prevent the gem loading entirely.